src/websites/services/domains.service.ts

```diff
@@ -32,50 +50,59 @@ export class DomainsService extends createPrismaBase(MODELS.Domain) {
   }
 
   async getDomainDetails(domainName: string) {
-    return this.route53.getDomainDetails(domainName)
+    return this.vercel.getDomainDetails(domainName)
   }
 
   async searchForDomain(domainName: string) {
-    const [availabilityResp, suggestionsResp, allPricesResp] =
-      await Promise.all([
-        this.route53.checkDomainAvailability(domainName),
-        this.route53.getDomainSuggestions(domainName),
-        this.route53.listPrices(),
-      ])
-
-    const allPricesMap = new Map()
-    allPricesResp?.Prices?.forEach((price) => {
-      if (price.Name && price.RegistrationPrice?.Price) {
-        allPricesMap.set(price.Name, {
-          registrationPrice: price.RegistrationPrice.Price,
-          renewalPrice: price.RenewalPrice?.Price,
-        })
-      }
-    })
-
-    const searchedTld = domainName.split('.').at(-1)
-    const searchedTldPrices = allPricesMap.get(searchedTld)
+    // Use AWS Route53 for domain availability and suggestions, but Vercel for pricing
+    const [availabilityResp, suggestionsResp] = await Promise.all([
+      this.route53.checkDomainAvailability(domainName),
+      this.route53.getDomainSuggestions(domainName),
+    ])
+
+    // Get pricing from Vercel for the main domain
+    let searchedDomainPrice: number | undefined
+    try {
+      const vercelPrice = await this.vercel.checkDomainPrice(domainName)
+      searchedDomainPrice = vercelPrice.price
+    } catch (error) {
+      this.logger.warn(`Could not get Vercel price for ${domainName}:`, error)
+    }
 
     const suggestions = suggestionsResp.SuggestionsList || []
-    const suggestionsWithPrices = suggestions.map((suggestion) => {
-      const suggestionTld = suggestion.DomainName?.split('.').at(-1)
-      const prices = allPricesMap.get(suggestionTld)
-
-      return {
-        ...suggestion,
-        prices: {
-          registration: prices?.registrationPrice,
-          renewal: prices?.renewalPrice,
-        },
-      }
-    })
+    const suggestionsWithPrices = await Promise.all(
+      suggestions.map(async (suggestion) => {
+        let price: number | undefined
+        try {
+          if (suggestion.DomainName) {
+            const vercelPrice = await this.vercel.checkDomainPrice(
+              suggestion.DomainName,
+            )
+            price = vercelPrice.price
+          }
+        } catch (error) {
+          this.logger.warn(
+            `Could not get Vercel price for ${suggestion.DomainName}:`,
+            error,
+          )
+        }
+
+        return {
+          ...suggestion,
+          prices: {
+            registration: price,
+            renewal: price,
+          },
+        }
+      }),
+    )
 
     return {
       domainName,
       availability: availabilityResp.Availability,
       prices: {
-        registration: searchedTldPrices?.registrationPrice,
-        renewal: searchedTldPrices?.renewalPrice,
+        registration: searchedDomainPrice,
+        renewal: searchedDomainPrice,
```

Comment:
Why not just return `price` property here w/ either `price` or `searchedDomainPrice` here? 

Why the two properties, when there's ever just one value, `price`, for a given request? 🤔